src/tests/end2end/BasicTests.cpp

```diff
@@ -35,8 +35,9 @@ TEST_P(BasicTests, BufferSetSubData) {
 }
 
 TEST_P(BasicTests, ReadPixelsTest) {
-    RGBA8 red(255, 0, 0, 255);
-    nxt::Buffer buffer = utils::CreateFrozenBufferFromData(device, &red, sizeof(red), nxt::BufferUsageBit::TransferSrc);
+    RGBA8 data[64];
+    data[0] = RGBA8(255, 0, 0, 255);
+    nxt::Buffer buffer = utils::CreateFrozenBufferFromData(device, data, sizeof(data), nxt::BufferUsageBit::TransferSrc);
```

Comment:
I don't know about the other backends, but D3D12 requires the buffer to be at least as large as the placed footprint. In validation, `ComputeTextureCopyBufferSize` now computes the buffer size as `rowPitch * location.height * location.depth`